packages/sonarwhal/src/lib/connectors/utils/requester.ts

```diff
@@ -177,6 +177,72 @@ export class Requester {
         return this._redirects.calculate(uri);
     }
 
+    /**
+     * Performs a `HEAD` request to the given `uri`
+     */
+    public head(uri: string): Promise<NetworkData> {
+        debug(`Requesting(HEAD) ${uri}`);
+
+        return new Promise((resolve: Function, reject: Function) => {
+            let rawBodyResponse: Buffer;
```

Comment:
All this method is very similar to the get one. Maybe there's a way to refactor it? Is the only difference that we return a body in one of them?

Reply:
I removed the head method from Requester. Found that i can initialize the Requester object with method set as `HEAD` in the options. Switched to that approach.